### PR TITLE
Switch the docs to HTML

### LINKED ISSUE #194: Use HTML renderer for docs
Now that #171 is merged it would be good to start using it for our own docs. @mortenpi would you like to do the honours?

### PR DESCRIPTION
**Pull request by mortenpi**
Resolves #194.

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -3,9 +3,47 @@ using Documenter
 makedocs(
     modules = [Documenter],
     clean   = false,
+    format   = Documenter.Formats.HTML,
+    sitename = "Documenter.jl",
+    pages    = [( # Compat: parens needed for 0.4 syntax compat
```
Comment (MichaelHatherly) [inline docs/make.jl]: I think using `Any[` should work shouldn't it?

**Comment by mortenpi** [comment]
Yup, `Any` looks better. I would still keep one of the `Compat` notes, since on 0.5+ only the `Any` syntax should probably be discouraged.

**Comment by mortenpi** [comment]
As I was adding a test to build the `test/examples` with HTML, I discovered that I had missed the `EvalNode` in `HTMLWriter`. So added the necessary function and also the test.

**Comment by MichaelHatherly** [comment]
LGTM, merge when ready.

> I discovered that I had missed the `EvalNode` in `HTMLWriter`. So added the necessary function and also the test.

:+1:

**Pull request merged**